static/js/containers/DashboardPage.js

```diff
@@ -229,6 +237,36 @@ class DashboardPage extends React.Component {
     }
   };
 
+  handleCoupon = () => {
+    const { coupons, dispatch, location: { query } } = this.props;
+
+    if (!query.coupon) {
+      // If there's no coupon code in the URL query parameters,
+      // there's nothing to do.
+      return;
+    }
+
+    if ( coupons.fetchPostStatus !== undefined ) {
+      // If we've already launched a coupon fetch, don't launch another one
+      return;
+    }
+
```

Comment:
Can you add an additional `if` statement here? We should wait until the original GET will succeed before doing the POST to prevent a race condition. Can you return if `coupons.fetchGetStatus !== FETCH_SUCCESS`?